exes/mqtt-broadcaster/src/impl.cpp

```diff
@@ -0,0 +1,118 @@
+#include "impl.hpp"
+
+namespace tfc::mqtt::impl {
+
+// Create a timestamp in milliseconds from 1 Jan 1970 according to spec
+auto timestamp_milliseconds() -> std::chrono::milliseconds {
+  using std::chrono::duration_cast;
+  using std::chrono::milliseconds;
+  using std::chrono::system_clock;
+  return duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+}
```

Comment:
Maybe just copy paste this code to its usage and remove this function, you probably do not need to make a duration cast